llvm/test/Transforms/LoopVectorize/optsize.ll

```diff
@@ -449,7 +411,7 @@ for.cond.cleanup28:
 
 for.body29:
   %i24.0170 = phi i16 [ 0, %entry], [ %inc37, %for.body29]
-  %add33 = add i16 undef, %i24.0170
```

Comment:
It looks like this removes a SCEV check in the NPGSO check lines even though it was branching on undef. It looks like this was originally testing that we don't assert when emitting a SCEV check, can we preserve that if we plumb through a function argument instead?